File: 728-self-dividing-numbers/self-dividing-numbers.js
var selfDividingNumbers = function(left, right) {
    let output = [];
    for (let i = left; i <= right; i++){
        if (i < 10)
        output.push(i);
        else{
            let isValid = true;
            let n = i;
        while(n>0){
            let d = n % 10;
            if ( d == 0 || i % d != 0)
            isValid=false;
            n = Math.floor(n / 10);
        }
        if (isValid)
        output.push(i);
    }
    }
    return output;
    
};

File: 728-self-dividing-numbers/self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        output = []
        for i in range(left, right+1):
            if i < 10:
                output.append(i)
            else:
                num_str = str(i)
                isValid = True
                for each in num_str:
                    each = int(each)
                    if each == 0 or i % each != 0:
                        isValid = False
                if isValid:
                    output.append(i)
        return output